overlord/snapstate/snapstate.go

```diff
@@ -122,7 +136,7 @@ func doInstall(s *state.State, snapst *SnapState, ss *SnapSetup) (*state.TaskSet
 	}
 
 	// copy-data (needs stopped services by unlink)
-	if !revisionIsLocal {
+	if !ss.IsRevert() {
```

Comment:
Thanks, makes a lot of sense. We were inferring revert vs. refresh from the simple presence, which is of course bogus.